src/ctapipe/instrument/optics.py

```diff
@@ -263,3 +269,161 @@ def __repr__(self):
 
     def __str__(self):
         return self.name
+
+
+class PSFModel:
+    def __init__(self, **kwargs):
+        """
+        Base component to describe image distortion due to the optics of the different cameras.
+        """
+
+    @classmethod
+    def subclass_from_name(cls, name, **kwargs):
+        """
+        Obtain an instance of a subclass via its name
+
+        Parameters
+        ----------
+        name : str
+            Name of the subclass to obtain
+
+        Returns
+        -------
+        Instance
+            Instance of subclass to this class
+        """
+
+        subclasses = {base.__name__: base for base in non_abstract_children(cls)}
+
+        requested_subclass = subclasses[name]
+        return requested_subclass(**kwargs)
+
+    @abstractmethod
+    def pdf(self, *args):
+        pass
+
+    @abstractmethod
+    def update_location(self, *args):
+        pass
+
+    @abstractmethod
+    def update_model_parameters(self, *args):
+        pass
+
+
+class ComaModel(PSFModel):
+    """
+    PSF model, describing pure coma aberrations PSF effect
+    """
+
+    def __init__(
+        self,
+        asymmetry_params=[0.49244797, 9.23573115, 0.15216096],
```

Comment:
Where do these magic numbers come from?

How are you expecting we determine these numbers? Is there a tool to compute them from a series of PSF images or pedestal images?

How do you expect them to be fed into ctapipe for the telescopes?

Reply:
I got these numbers from https://gitlab.cern.ch/cta-unige/startracker (line [162 https://gitlab.cern.ch/cta-unige/startracker/-/blob/master/startracker/analysis/image.py?ref_type=heads]). @mexanick generated these parameters based on some fits. I can remove these as default.

Reply:
> So, when i will use this for extracting stars these parameters will not need to be altered

*for LST-1*, right?

Reply:
Ah, yes, they need to be configured per telescope.

Reply:
Ok, then I think we should discuss a bit about API.

I think, it's natural then to make this a `TelescopeComponent` and if there is code that can take a list of images and fit the result, we could have that as a 
```
@classmethod
def fit(self, ...) -> Parameters: 
```
here

Reply:
that makes sense. For the fitting though my idea was to fit the PSF based on images with multiple stars. I have made some draft of a function that would generate a picture based on star locations, magnitudes and psf parameters that would be used as a model for fitting: 

`def get_star_pdf(r0, f0, geometry, psf, n_pdf_bins, pdf_bin_size, focal_length):
    image = np.zeros(len(geometry))

    dr = pdf_bin_size * np.rad2deg(np.arctan(1 / focal_length.to_value(u.m))) / 3600.0
    r = np.linspace(
        r0 - dr * n_pdf_bins / 2.0,
        r0 + dr * n_pdf_bins / 2.0,
        n_pdf_bins,
    )
    df = np.deg2rad(pdf_bin_size / 3600.0) * 100
    f = np.linspace(
        f0 - df * n_pdf_bins / 2.0,
        f0 + df * n_pdf_bins / 2.0,
        n_pdf_bins,
    )

    for r_ in r:
        for f_ in f:
            val = psf.pdf(r_, f_, r0, f0) * dr * df
            x, y, _ = r_ * s2c(f_, 0)
            pixelN = geometry.position_to_pix_index(x * u.m, y * u.m)
            if pixelN > -1 and pixelN < len(image):
                image[pixelN] += val

    return image

def StarImageGenerator(
    radius,
    phi,
    magnitude,
    n_pdf_bins,
    pdf_bin_size,
    psf_model,
    camera_geometry,
    focal_length,
):
    """
    :param list stars: list of star containers, stars to be placed in image
    :param dict psf_model_pars: psf model parameters
    """

    image = np.zeros(len(camera_geometry))
    for r, p, m in zip(radius, phi, magnitude):
        image += np.power(10,-0.4*m) * get_star_pdf(
            r, p, camera_geometry, psf_model, n_pdf_bins, pdf_bin_size, focal_length
        )

    return image`

Reply:
I made the PSFModel into a `TelescopeComponent`. I will need to see how i will handle the fitting in detail, but i can work that out.